quinn-proto/src/connection.rs

```diff
@@ -745,16 +745,52 @@ impl Connection {
         remote: SocketAddrV6,
         partial_decode: PartialDecode,
     ) -> Option<BytesMut> {
-        let crypto = if partial_decode.is_handshake() {
-            &self.handshake_crypto
+        let (result, crypto_update) = if partial_decode.is_handshake() {
+            (
+                partial_decode.finish(self.handshake_crypto.pn_decrypt_key()),
+                None,
+            )
+        } else if partial_decode.key_phase() == self.key_phase {
+            if let Some(ref crypto) = self.crypto.as_ref() {
+                (partial_decode.finish(&crypto.pn_decrypt_key()), None)
+            } else {
+                warn!(
+                    self.log,
+                    "recieved a non-hanshake packet before crypto establishment"
```

Comment:
```suggestion
                    "recieved a non-hanshake packet before handshake completed"
```

Reply:
(Note, you probably also want to replace `hanshake` with `handshake`.)